### PR TITLE
Fix: update natspec - issue 91

### PR DESCRIPTION
**Pull request by Jean-Grimal**
Fixes https://github.com/cantinasec/review-morpho-blue-1/issues/91

### CODE REVIEW

File: src/MetaMorpho.sol [hunk 1]
```diff
@@ -280,8 +280,8 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /* ONLY ALLOCATOR FUNCTIONS */
 
     /// @notice Sets `supplyQueue` to `newSupplyQueue`.
-    /// @dev The supply queue can be a set containing duplicate markets, but it would only increase the cost of
-    /// depositing to the vault.
+    /// @dev `newSupplyQueue` is an array of enabled markets, and can contain duplicate markets, but it would only
+    /// increase the cost of depositing to the vault.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:284]: Let's `@param` no?

File: src/MetaMorpho.sol [hunk 2]
```diff
@@ -296,13 +296,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         emit EventsLib.SetSupplyQueue(_msgSender(), newSupplyQueue);
     }
 
-    /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with zero cap and zero
-    /// vault's supply can be removed.
+    /// @notice Sets `WithdrawQueue` as a new withdraw queue defined by `indexes`.
+    /// @notice This is the only entry point to disable a market.
+    /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with both zero cap and
+    /// zero vault's supply can be removed from the permutation to be disabled.
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:302]:
```suggestion
    /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with both zero cap and
    /// zero vault's supply can be removed from the permutation to be disabled.
    /// @notice This is the only entry point to disable a market.
```

File: src/MetaMorpho.sol [hunk 3]
```diff
@@ -296,13 +296,15 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
...
     /// @notice Removing a market requires the vault to have 0 supply on it; but anyone can supply on behalf of the
     /// vault so the call to `sortWithdrawQueue` can be griefed by a frontrun. To circumvent this, the allocator can
     /// simply bundle a reallocation that withdraws max from this market with a call to `sortWithdrawQueue`.
     /// @param indexes The indexes of each market in the previous withdraw queue, in the new withdraw queue's order.
-    function sortWithdrawQueue(uint256[] calldata indexes) external onlyAllocatorRole {
+    function setWithdrawQueue(uint256[] calldata indexes) external onlyAllocatorRole {
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:307]: I understand why `sort` is not best-suited
I also think that `set` is less suited

What should we do?

File: src/MetaMorpho.sol [hunk 4]
```diff
@@ -759,10 +761,6 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             Id id = withdrawQueue[i];
             MarketParams memory marketParams = _marketParams(id);
 
-            // The vault withdrawing from Morpho cannot fail because:
-            // 1. oracle.price() is never called (the vault doesn't borrow)
-            // 2. `_withdrawable` caps to the liquidity available on Morpho
-            // 3. virtually accruing interest didn't fail in `_withdrawable`
```
Comment (Rubilmax) [inline src/MetaMorpho.sol:765]: See why I disagree with the finding: https://github.com/cantinasec/review-morpho-blue-1/issues/91#issuecomment-1790887251

**Review by Rubilmax — CHANGES_REQUESTED** [review]

File: src/MetaMorpho.sol (hunk 3 above)
Comment (Jean-Grimal) [reply src/MetaMorpho.sol:307]: As markets can be removed, this function doesn't always just "sort" the queue, the the issue requested to change the naming.

Comment (Rubilmax) [reply src/MetaMorpho.sol:306]: Yes I understand but still, I think that `set` is less suited and even misleading compared to `sort`

Comment (Rubilmax) [reply src/MetaMorpho.sol:306]: Great!

**Review by Rubilmax — APPROVED** [review]

File: src/MetaMorpho.sol [hunk 5]
```diff
@@ -296,13 +296,14 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         emit EventsLib.SetSupplyQueue(_msgSender(), newSupplyQueue);
     }
 
-    /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with zero cap and zero
-    /// vault's supply can be removed.
+    /// @notice Sets the withdraw queue as a permutation of the previous one, although markets with both zero cap and
+    /// zero vault's supply can be removed from the permutation to be disabled.
```
Comment (MerlinEgalite) [inline src/MetaMorpho.sol:300]:
```suggestion
    /// zero vault's supply can be removed from the permutation.
```

**Review by Rubilmax — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**